### PR TITLE
Adds Attach Screenshot Option

### LINKED ISSUE #2362: Attach screenshot
Add option `attachScreenshot` that when opt in, attached a screenshot on error events.

### PR DESCRIPTION
**Pull request by PaitoAnderson**
Fixes #2362

## :loudspeaker: Type of change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring

## :scroll: Description

Adds a `attachScreenshot` option that sets the `attachScreenshot` in cocoa project and calls `setAttachScreenshot` in java project. 

## :bulb: Motivation and Context

Excited about using the new Sentry screenshot feature in React Native projects.

## :green_heart: How did you test it?
Ran the sample projects.

iOS:
image

Android:
image
(Not sure what I have to do to trigger a screenshot, didn't work for me, but did verify the setting was indeed set at init)

## :pencil: Checklist

- [x] I reviewed submitted code
- [ ] I added tests to verify changes
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## Unreleased
+
+- feat: Adds `attachScreenshot` option.
```
Comment (marandaneto) [inline CHANGELOG.md:5]:
```suggestion
### Features

- feat: Adds `attachScreenshot` option.
```

**Comment by marandaneto** [comment]
Thanks @PaitoAnderson 
I'll check what's going on on Android.

**Comment by marandaneto** [comment]
> Thanks @PaitoAnderson I'll check what's going on on Android.

https://github.com/getsentry/sentry-react-native/issues/2362#issuecomment-1193763553

**Comment by PaitoAnderson** [comment]
Thanks @marandaneto for all your work figuring out why this didn't work as I had hoped; based on the changes that need to happen before this relatively simple PR can be merged would you rather I closed this PR or leave it open?

**Comment by marandaneto** [comment]
Replaced by https://github.com/getsentry/sentry-react-native/pull/2610

**Pull request closed without merging**